src/Controller/UserRefreshCodeController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Controller;
+
+use App\Exceptions\CodeGenerationException;
+use App\Repository\UserActivationCodeRepository;
+use App\Service\ExternalUserService;
+use App\Utils\ValidationUtils;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\AsController;
+
+#[AsController]
+class UserRefreshCodeController extends AbstractController
+{
+    public function __construct(
+        private readonly UserActivationCodeRepository $activationCodeRepository,
+        private readonly ExternalUserService $externalUserService,
+        private readonly ValidationUtils $validationUtils,
+    ) {}
+
+    /**
+     * @throws CodeGenerationException
```

Comment:
This should be caught and converted to the relevant `HttpException`

Reply:
Fixed